sympy/core/tests/test_power.py

```diff
@@ -452,3 +452,20 @@ def test_better_sqrt():
     assert sqrt(-I/2) == Mul(S.Half, 1 - I, evaluate=False)
     # fractional im part
     assert Pow(-9*I/2, 3/S(2)) == 27*(1 - I)**3/8
+
+def test_idempotence():
+    x = Symbol('x', idempotent=True)
+    n = Symbol('n', positive=True)
+    y = Symbol('y')
+
+    # positive powers equal the element
+    assert x**2     == x
+    assert x**125   == x
+    assert x**n     == x
+    assert x**(n+1) == x
+
+    # empty product
+    assert x**0     == 1
+
+    # not-necessarily-positive powers are left untouched
+    assert x**y     != x
```

Comment:
Can you assert what this is equal to e.g. `Pow(x, y, evaluate=False)`?

Reply:
Done.